From 74f34103f513e8718c27f853132c633c3045ee44 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marek.vasut+renesas@mailbox.org>
Date: Thu, 20 Mar 2025 04:31:15 +0100
Subject: [PATCH] Draft: FIXME: Force DSC clock on

Enable DSC clock to oeprate DSC in bypass and get video out of DSI1/DP.

Signed-off-by: Marek Vasut <marek.vasut+renesas@mailbox.org>
---
 drivers/clk/renesas/r8a779g0-cpg-mssr.c | 5 +++++
 drivers/clk/renesas/renesas-cpg-mssr.c  | 4 ++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/renesas/r8a779g0-cpg-mssr.c b/drivers/clk/renesas/r8a779g0-cpg-mssr.c
index 8f2ea58bfdee..be4c98b5ffb7 100644
--- a/drivers/clk/renesas/r8a779g0-cpg-mssr.c
+++ b/drivers/clk/renesas/r8a779g0-cpg-mssr.c
@@ -368,6 +368,7 @@ static int __init r8a779g0_cpg_mssr_init(struct device *dev)
 
 static const unsigned int r8a779g0_crit_mod_clks[] __initconst = {
 	MOD_CLK_ID(330),	/* csdbgpap */
+	MOD_CLK_ID(2819),	/* DSC */
 };
 
 const struct cpg_mssr_info r8a779g0_cpg_mssr_info __initconst = {
@@ -386,6 +387,10 @@ const struct cpg_mssr_info r8a779g0_cpg_mssr_info __initconst = {
 	.crit_mod_clks = r8a779g0_crit_mod_clks,
 	.num_crit_mod_clks = ARRAY_SIZE(r8a779g0_crit_mod_clks),
 
+	/* Critical Module Clocks */
+	.crit_mod_clks = r8a779g0_crit_mod_clks,
+	.num_crit_mod_clks = ARRAY_SIZE(r8a779g0_crit_mod_clks),
+
 	/* Callbacks */
 	.init = r8a779g0_cpg_mssr_init,
 	.cpg_clk_register = rcar_gen4_cpg_clk_register,
diff --git a/drivers/clk/renesas/renesas-cpg-mssr.c b/drivers/clk/renesas/renesas-cpg-mssr.c
index ce023a463890..710ced509bfc 100644
--- a/drivers/clk/renesas/renesas-cpg-mssr.c
+++ b/drivers/clk/renesas/renesas-cpg-mssr.c
@@ -455,8 +455,8 @@ static void __init cpg_mssr_register_mod_clk(const struct mssr_mod_clk *mod,
 	clock->hw.init = &init;
 
 	for (i = 0; i < info->num_crit_mod_clks; i++)
-		if (id == info->crit_mod_clks[i] &&
-		    cpg_mstp_clock_is_enabled(&clock->hw)) {
+		if (id == info->crit_mod_clks[i]/* &&
+		    cpg_mstp_clock_is_enabled(&clock->hw)*/) {
 			dev_dbg(dev, "MSTP %s setting CLK_IS_CRITICAL\n",
 				mod->name);
 			init.flags |= CLK_IS_CRITICAL;
-- 
2.34.1

